# HG changeset patch
# Parent 3265a3936a0f7450ef2161fd09d1d0340ab927f8
Use the generic implementation of libstdc++ primitives when we're on musl, not the glibc one.

diff -r 3265a3936a0f libstdc++-v3/configure.host
--- a/libstdc++-v3/configure.host	Tue May 27 17:11:50 2014 -0400
+++ b/libstdc++-v3/configure.host	Tue May 27 17:18:49 2014 -0400
@@ -264,6 +264,13 @@
     os_include_dir="os/bsd/freebsd"
     ;;
   gnu* | linux* | kfreebsd*-gnu | knetbsd*-gnu)
+    # check for musl by target
+    case "${host_os}" in
+      *-musl*)
+        os_include_dir="os/generic"
+        ;;
+      *)
+
     if [ "$uclibc" = "yes" ]; then
       os_include_dir="os/uclibc"
     elif [ "$bionic" = "yes" ]; then
@@ -272,6 +279,9 @@
       os_include_dir="os/gnu-linux"
     fi
     ;;
+
+    esac
+    ;;
   hpux*)
     os_include_dir="os/hpux"
     ;;
